src/wallet/wallet2.cpp

```diff
@@ -1602,6 +1603,48 @@ std::string wallet2::get_subaddress_label(const cryptonote::subaddress_index& in
   return m_subaddress_labels[index.major][index.minor];
 }
 //----------------------------------------------------------------------------------------------------
+void wallet2::scan_tx(const std::set<std::string> txids)
+{
+  // Get the transactions from daemon in batches and add them to a priority queue ordered in chronological order
+  auto cmp_tx_entry = [](cryptonote::COMMAND_RPC_GET_TRANSACTIONS::entry& l, cryptonote::COMMAND_RPC_GET_TRANSACTIONS::entry& r)
+  { return l.block_timestamp > r.block_timestamp; };
```

Comment:
Use block height (timestamps are not monotonic). There does not seem to be a foolproof way to sort within a block as a tie breaker but I don't think this should create issues in practice.